extension/src/mock.rs

```diff
@@ -4,13 +4,42 @@ use crate::{
 };
 use codec::{Decode, Encode};
 use frame_support::{
-	derive_impl, pallet_prelude::Weight, parameter_types, traits::ConstU32, traits::Everything,
+	derive_impl,
+	pallet_prelude::Weight,
+	parameter_types,
+	traits::{ConstU32, Everything, Nothing},
 };
 use frame_system::pallet_prelude::BlockNumberFor;
 use pallet_contracts::{chain_extension::RetVal, DefaultAddressGenerator, Frame, Schedule};
 use sp_runtime::Perbill;
 use std::marker::PhantomData;
 
+type DispatchCallWith<Id, Filter> = DispatchCall<
+	// Registered with func id 1
+	WithFuncId<Id>,
+	// Runtime config
+	Test,
+	// Decode inputs to the function as runtime calls
+	Decodes<RuntimeCall, DecodingFailed<Test>, RemoveFirstByte>,
+	// Accept any filterting
+	Filter,
+>;
+
+type ReadStateWith<Id, Filter> = ReadState<
+	// Registered with func id 1
+	WithFuncId<Id>,
+	// Runtime config
+	Test,
+	// The runtime state reads available.
+	RuntimeRead,
+	// Decode inputs to the function as runtime calls
+	Decodes<RuntimeRead, DecodingFailed<Test>, RemoveFirstByte>,
+	// Accept any filtering
+	Filter,
+	// Convert the result of a read into the expected result
+	DefaultConverter<RuntimeResult>,
+>;
+
```

Comment:
I like this. Can you use this in `type Functions` in this file and use it for all test cases?

Reply:
Yes it is currently used.

Reply:
Ah shoot, nice!